src/core/CoolBoard.cpp

```diff
@@ -377,11 +378,71 @@ void CoolBoard::readBoardData(JsonObject &reported) {
   }
 }
 
-void CoolBoard::sleep(unsigned long interval) {
-  if (interval > 0) {
-    INFO_VAR("Going to sleep for seconds:", interval);
-    ESP.deepSleep((uint64(interval) * 1000000ULL), WAKE_RF_DEFAULT);
+void CoolBoard::shiftLoginterval(uint32_t value) {
+  uint8_t val[4];
+  val[0] = value - MAX_SLEEP_TIME;
+  val[1] = ((value - MAX_SLEEP_TIME) >> 8);
+  val[2] = ((value - MAX_SLEEP_TIME) >> 16);
+  val[3] = ((value - MAX_SLEEP_TIME) >> 24);
```

Comment:
you can do a for  to group bitshift in 3 lines

Reply:
Agreed